roxie/ccd/ccdmain.cpp

```diff
@@ -1525,7 +1525,8 @@ int CCD_API roxie_main(int argc, const char *argv[], const char * defaultYaml)
                             const char *certIssuer = roxieFarm.queryProp("@issuer");
                             if (isEmptyString(certIssuer))
                                 certIssuer = roxieFarm.getPropBool("@public", true) ? "public" : "local";
-                            tlsConfig.setown(getIssuerTlsSyncedConfig(certIssuer, roxieFarm.queryProp("trusted_peers"), false));
+                            bool disableMtls = roxieFarm.getPropBool("@disableMtls", false);
+                            tlsConfig.setown(getIssuerTlsSyncedConfig(certIssuer, roxieFarm.queryProp("trusted_peers"), disableMtls));
```

Comment:
@ghalliday Considering this is for issuer based config, I'm curious what use case / issuer scenario this is needed for?

As is, without this, public should have mtls effectively turned off, remote and local should have it turned on.
The only scenario that comes to mind is local to components in this environment, but without mtls?

Perhaps this is for a bare-metal scenario, but I'm still not sure what the details would be.

Reply:
We discussed this offline.  In bare-metal multiple components can be configured to run on the same hardware.  Different issuer names may be used to distinguish the issuer secret locations.  Therefore we may not be able to rely on the issuer name of "public".  disableMtls would be used to turn off client certificates for a public issuer that had been given a "mangled" name.